### PR TITLE
Nano Server tutorial

### PR DESCRIPTION
**Pull request by shirhatti**
Addresses #194 

@danroth27 Still needs work on the language, but I want you to have an initial look and see if I captured what you wanted.

### CODE REVIEW

File: aspnet/tutorials/nano-server.rst [hunk 1]
```diff
@@ -1,8 +1,149 @@
-.. include:: /../common/stub-topic.txt
+.. _nano-server:
+
+ASP.NET 5 on Nano Server
+========================
+
+By `Sourabh Shirhatti`
+
+.. attention:: This tutorial uses a pre-lease version of the Nano Server installation option of Windows Server Technical Preview 4. You may use the software in the virtual hard disk image only to internally demonstrate and evaluate it. You may not use the software in a live operating environment. Please see https://go.microsoft.com/fwlink/?LinkId=624232 for specific information about the end date for the software.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: What does "internally" mean in this context?

Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > end date for the software

end date for this preview.

File: aspnet/tutorials/nano-server.rst [hunk 2]
```diff
@@ -1,8 +1,149 @@
...
+In this tutorial, you'll take an existing ASP.NET 5 application and deploy it to a Nano Server instance running IIS.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Introduction
+------------
+
+Nano Server is a new version of Windows Server with a small footprint optimized for the cloud and a DevOps workflow. You can learn more about Nano Server from the official docs `available here <https://msdn.microsoft.com/en-us/library/mt126167.aspx>`_.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Not sure what "DevOps workflow" really means here.

Comment (shirhatti) [inline aspnet/tutorials/nano-server.rst]: Borrowed the language from [here](https://blogs.technet.microsoft.com/windowsserver/2015/04/08/microsoft-announces-nano-server-for-modern-apps-and-cloud/). I can drop it

**Comment by danroth27** [comment]
The structure looks fine.

@rachelappel @Rick-Anderson Please review.

File: aspnet/tutorials/nano-server.rst [hunk 3]
```diff
@@ -1,8 +1,149 @@
...
+.. _nano-server:
+
+ASP.NET 5 on Nano Server
+========================
+
+By `Sourabh Shirhatti`
+
+.. attention:: This tutorial uses a pre-lease version of the Nano Server installation option of Windows Server Technical Preview 4. You may use the software in the virtual hard disk image only to internally demonstrate and evaluate it. You may not use the software in a live operating environment. Please see https://go.microsoft.com/fwlink/?LinkId=624232 for specific information about the end date for the preview.
+
+In this tutorial, you'll take an existing ASP.NET 5 application and deploy it to a Nano Server instance running IIS.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: I would borrow some more text from the intro paragraph at https://msdn.microsoft.com/en-us/library/mt126167.aspx to motivate the benefits of Nano Server.

File: aspnet/tutorials/nano-server.rst [hunk 4]
```diff
@@ -1,8 +1,149 @@
...
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Introduction
+------------
+
+Nano Server is a new version of Windows Server with a small footprint optimized for the cloud. You can learn more about Nano Server from the official docs `available here <https://msdn.microsoft.com/en-us/library/mt126167.aspx>`_.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Avoid the "click here" pattern.

> You can learn more about Nano Server from the official docs `` `available here <https://msdn.microsoft.com/en-us/library/mt126167.aspx>`_``.

You can learn more about Nano Server from the `` `official docs <https://msdn.microsoft.com/en-us/library/mt126167.aspx>`_.``

File: aspnet/tutorials/nano-server.rst [hunk 5]
```diff
@@ -1,8 +1,149 @@
...
+
+In this tutorial, we will using a pre-built Virtual Hard Disk (VHD) for Nano Server which can be downloaded `here <https://msdn.microsoft.com/en-us/virtualization/windowscontainers/nano_eula>`_. This pre-built VHD already includes the Reverse Forwarders and IIS packages which are required for this tutorial.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > In this tutorial, we will using a pre-built Virtual Hard Disk (VHD) for Nano Server which can be downloaded `here <https://msdn.microsoft.com/en-us/virtualization/windowscontainers/nano_eula>`_. 

In this tutorial, we will using the pre-built `` `Virtual Hard Disk (VHD) for Nano Server <https://msdn.microsoft.com/en-us/virtualization/windowscontainers/nano_eula>_` `` from Windows Server Technical Preview 4.

File: aspnet/tutorials/nano-server.rst [hunk 6]
```diff
@@ -1,8 +1,149 @@
...
+
+Before proceeding with this tutorial, you will require the published output of an existing ASP.NET 5 application. Refer to the guidance on :ref:`dnu-publish`. Ensure your application is built targeting the **64-bit** version of ``coreclr``.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > you will require

you will need

Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Can you provide the one-liner that shows how to do the publish from the command line? `dnu publish -something blah blah`.

File: aspnet/tutorials/nano-server.rst [hunk 7]
```diff
@@ -1,8 +1,149 @@
...
+
+Setting up the Nano Server Instance
+-----------------------------------
+
+Create a new Virtual Machine using Hyper-V on your development machine using the previously downloaded VHD. The machine will require you to set an administator password before logging on. At the VM console, press F11 to set the password before the first logon.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: `` `Create a new Virtual Machine using Hyper-V <https://technet.microsoft.com/en-us/library/hh846766.aspx>`__``

File: aspnet/tutorials/nano-server.rst [hunk 8]
```diff
@@ -1,8 +1,149 @@
...
+
+After setting the local password, you will manage Nano Server using the management tools remotely.
+
+Connecting to your Nano Server Instance using PowerShell Remote
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: using PowerShell Remoting

File: aspnet/tutorials/nano-server.rst [hunk 9]
```diff
@@ -1,8 +1,149 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open an elevated PowerShell window to add your remote Nano Server instance to your ``TrustedHosts`` list.
+
+.. code:: ps1
+
+  $ip = "10.83.181.14" # replace with the correct IP address
+  Set-Item WSMan:\localhost\Client\TrustedHosts "$ip" -Force
+
+Once you have added your Nano Server instance to your ``TrustedHosts``, you can connect to it using PowerShell Remote session
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > using PowerShell Remote session

using PowerShell remoting.

File: aspnet/tutorials/nano-server.rst [hunk 10]
```diff
@@ -1,8 +1,149 @@
...
+
+.. code:: ps1
+
+  $ip = "10.83.181.14" # replace with the correct IP address
+  $s = New-PSSession -ComputerName $ip -Credential ~\Administrator
+  Enter-PSSession $s
+
+If you have successfully connected using PSRemote, you prompt will look like this ``[10.83.181.14]: PS C:\Users\Administrator\Documents>``
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > If you have successfully connected using PSRemote, you prompt will look like this 

If you have successfully connected then your prompt will look like this

File: aspnet/tutorials/nano-server.rst [hunk 11]
```diff
@@ -1,8 +1,149 @@
...
+
+Installing the HTTPPlatformHandler Module
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: The casing should HttpPlatformHandler.

File: aspnet/tutorials/nano-server.rst [hunk 12]
```diff
@@ -1,8 +1,149 @@
...
+-----------------------------------------
+
+At the moment, the process to install the HTTPPlatformHandler Module for IIS is manual. You will need to install the latest 64-bit version of the `HttpPlatformHandler <http://www.iis.net/downloads/microsoft/httpplatformhandler>`_ on a regular (not Nano) machine. After installing you will need to copy the following files:
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Add a sentence or two stating what the HttpPlatformHandler is and why it is needed.

File: aspnet/tutorials/nano-server.rst [hunk 13]
```diff
@@ -1,8 +1,149 @@
...
+
+.. note:: This script runs on a clean system, but is not meant to be idempotent. If you run this multiple times it will add multiple entries. If you end up in a bad state, you can find backups of the *applicationHost.config* file at *%systemdrive%\inetpub\history*.
+
+.. literalinclude:: nano-server/enable-platformhandler.ps1
+  :language: ps1
+
+Manually Editing *applicationHost.config*
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can skip this section if you already ran the PowerShell script above. Though is not recommended, you can also enable the HTTPPlatformHandler by manually editing the *applicationHost.config* file.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > you can also enable

> you can alternatively enable

File: aspnet/tutorials/nano-server.rst [hunk 14]
```diff
@@ -1,8 +1,149 @@
...
+
+Open up *C:\\Windows\\System32\\inetsrv\\config\\applicationHost.config*
+
+Under ``<configSections>`` add
+
+.. literalinclude:: nano-server/applicationHost.config
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: dedent

File: aspnet/tutorials/nano-server.rst [hunk 15]
```diff
@@ -1,8 +1,149 @@
...
+  :language: xml
+  :lines: 42-43
+  :emphasize-lines: 2
+
+In the ``system.webServer`` section, unlock handlers
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > In the ``system.webServer`` section, unlock handlers 

In the system.webServer section group, update the handlers section to allow the configured handlers to be overridden.

File: aspnet/tutorials/nano-server.rst [hunk 16]
```diff
@@ -1,8 +1,149 @@
...
+
+.. literalinclude:: nano-server/applicationHost.config
+  :language: xml
+  :lines: 55,63
+  :emphasize-lines: 2
+
+Add ``HTTPPlatformHandler`` to the ``globalModules`` section
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: httpPlatformHandler

File: aspnet/tutorials/nano-server.rst [hunk 17]
```diff
@@ -1,8 +1,149 @@
...
+
+.. literalinclude:: nano-server/applicationHost.config
+  :language: xml
+  :lines: 210,224
+  :emphasize-lines: 2
+
+Additionally, add ``HTTPPlatformHandler`` to the ``modules`` section
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: httpPlatformHandler

File: aspnet/tutorials/nano-server.rst [hunk 18]
```diff
@@ -1,8 +1,149 @@
...
+
+.. literalinclude:: nano-server/applicationHost.config
+  :language: xml
+  :lines: 275,286
+  :emphasize-lines: 2
+
+Installing an ASP.NET 5 application
+-----------------------------------
+
+I have copied the published contents of my application to *C:\\HelloAspNet5\\* on the Nano server. Let's create a new site in IIS to point to the output of our published app using this PowerShell snippet. We are using the ``Default App Pool`` in this example for simplicity. For more considerations on running under an AppPool, refer to the :ref:`apppool` doc.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Show the copy step.

Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > Let's create a new site in IIS to point to the output of our published app using this PowerShell snippet. We are using the ``Default App Pool`` in this example for simplicity. For more considerations on running under an AppPool, refer to the :ref:`apppool` doc. 

Use the following PowerShell snippet to create a new site in IIS for our published app. This script uses the ``Default App Pool`` for simplicity. For more considerations on running under an AppPool, see :ref:`apppool`.

**Comment by danroth27** [comment]
:watch:

@blowdart Can you take a look at this doc as well?

**Comment by blowdart** [comment]
Nope. I know nothing about nanoserver or it's security model, so my looking this over would be a rubber stamping exercise.

File: aspnet/tutorials/nano-server.rst (hunk 12 above)
Comment (shirhatti) [inline aspnet/tutorials/nano-server.rst]: Isn't it more appropriate to link to https://docs.asp.net/en/latest/hosting/http-platformhandler.html ?

**Comment by shirhatti** [comment]
@danroth27 Responded to your feedback

File: aspnet/tutorials/nano-server.rst [hunk 19]
```diff
@@ -1,8 +1,164 @@
...
+  :depth: 1
+
+Introduction
+------------
+
+Windows Server 2016 Technical Preview offers a new installation option: Nano Server. Nano Server is a remotely administered server operating system optimized for private clouds and datacenters. It takes up far less disk space, sets up significantly faster, and requires far fewer updates and restarts than Windows Server. You can learn more about Nano Server from the `official docs <https://msdn.microsoft.com/en-us/library/mt126167.aspx>`_.
+
+In this tutorial, we will be using the pre-built `Virtual Hard Disk (VHD) for Nano Server <https://msdn.microsoft.com/en-us/virtualization/windowscontainers/nano_eula>`_  from Windows Server Technical Preview 4. This pre-built VHD already includes the Reverse Forwarders and IIS packages which are required for this tutorial.
+
+Before proceeding with this tutorial, you will need the published output of an existing ASP.NET 5 application. Refer to the guidance on :ref:`publishing <dnu-publish>`. Ensure your application is built targeting the **64-bit** version of ``coreclr``.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > Before proceeding with this tutorial, you will need the published output of an existing ASP.NET 5 application. Refer to the guidance on :ref:`publishing <dnu-publish>`. Ensure your application is built targeting the **64-bit** version of ``coreclr``.

Before proceeding with this tutorial, you will need the :ref:`published <dnu-publish>` output of an existing ASP.NET 5 application. Ensure your application is published with the **64-bit** version of ``coreclr``.

File: aspnet/tutorials/nano-server.rst [hunk 20]
```diff
@@ -1,8 +1,164 @@
...
+.. code:: ps1
+
+  dnu publish --runtime dnx-coreclr-win-x64.1.0.0-rc1-update1
+
+Setting up the Nano Server Instance
+-----------------------------------
+
+`Create a new Virtual Machine using Hyper-V <https://technet.microsoft.com/en-us/library/hh846766.aspx>`_ on your development machine using the previously downloaded VHD. The machine will require you to set an administator password before logging on. At the VM console, press F11 to set the password before the first logon.
+
+After setting the local password, you will manage Nano Server using the management tools remotely.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > using the management tools remotely

using PowerShell remoting.

File: aspnet/tutorials/nano-server.rst [hunk 21]
```diff
@@ -1,8 +1,164 @@
...
+  $ip = "10.83.181.14" # replace with the correct IP address
+  $s = New-PSSession -ComputerName $ip -Credential ~\Administrator
+  Enter-PSSession $s
+
+If you have successfully connected then your prompt will look like this ``[10.83.181.14]: PS C:\Users\Administrator\Documents>``
+
+Installing the HttpPlatformHandler Module
+-----------------------------------------
+
+The :ref:`http-platformhandler` is an IIS 7.5+ module which is responsible for process management of http listeners and to proxy requests to processes that it manages. At the moment, the process to install the HttpPlatformHandler Module for IIS is manual. You will need to install the latest 64-bit version of the `HttpPlatformHandler <http://www.iis.net/downloads/microsoft/HttpPlatformHandler>`_ on a regular (not Nano) machine. After installing you will need to copy the following files:
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > The :ref:`http-platformhandler` is an IIS 7.5+ module

The :ref:`HttpPlatformHandler <http-platformhandler>` is an IIS 7.5+ module

File: aspnet/tutorials/nano-server.rst (hunk 20 above)
Comment (shirhatti) [inline aspnet/tutorials/nano-server.rst]: Not entirely true. There are remote management tools not limited to Powershell remoting

File: aspnet/tutorials/nano-server.rst [hunk 22]
```diff
@@ -1,8 +1,164 @@
...
+
+Copy over the published output of your existing application to the Nano server.
+
+.. code:: ps1
+
+  $ip = "10.83.181.14" # replace with the correct IP address
+  $s = New-PSSession -ComputerName $ip -Credential ~\Administrator
+  Copy-Item -ToSession $s -Path <path-to-src>\bin\output\* -Destination C:\HelloAspNet5 -Recurse
+
+Use the following PowerShell snippet to create a new site in IIS for our published app. This script uses the ``DefaultAppPool`` for simplicity. For more considerations on running under an AppPool, see :ref:`apppool`.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > an AppPool

an application pool

File: aspnet/tutorials/nano-server.rst [hunk 23]
```diff
@@ -1,8 +1,164 @@
...
+
+.. code:: ps1
+
+  New-NetFirewallRule -Name "AspNet5" -DisplayName "HTTP on TCP/8000" -Protocol TCP -LocalPort 8000 -Action Allow -Enabled True
+
+Running the Application
+-----------------------
+
+At this point your published web application, should be accessible in browser by visiting ``http://<ip-address>:8000``.
+If you have set up logging as described in :ref:`log-redirection`, you should be able to view your logs at *C:\\HelloAspNet5\\logs*.
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: > *C:\\HelloAspNet5\\logs*

This renders oddly. Maybe use double back-ticks? ``C:\\HelloAspNet5\\logs``

File: aspnet/tutorials/nano-server.rst [hunk 24]
```diff
@@ -1,8 +1,164 @@
-.. include:: /../common/stub-topic.txt
+.. _nano-server:
 
-|stub-icon| ASP.NET 5 on Nano Server
-====================================
+ASP.NET 5 on Nano Server
+========================
+
+By `Sourabh Shirhatti`
```
Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: Optional: update authors.txt and add a link for your name. Your github user works.

**Comment by danroth27** [comment]
A couple more minor nits, then :shipit:

File: aspnet/tutorials/nano-server.rst (hunk 24 above)
Comment (shirhatti) [inline aspnet/tutorials/nano-server.rst]: Hmm. That's weird. My name is already added to https://github.com/aspnet/Docs/blob/master/common/authors.txt#L13

Do you know why it doesn't render correctly?

File: aspnet/tutorials/nano-server.rst (hunk 23 above)
Comment (shirhatti) [inline aspnet/tutorials/nano-server.rst]: I'm already using double back-ticks.

Comment (danroth27) [inline aspnet/tutorials/nano-server.rst]: ??

You're using asterisks around the file path, not double back ticks.

**Pull request merged**